crates/sui-core/src/authority/consensus_tx_status_cache.rs

```diff
@@ -86,6 +113,10 @@ impl ConsensusTxStatusCache {
                     // FastpathCertified is transient and can be updated to other statuses.
                     (ConsensusTxStatus::FastpathCertified, _) => {
                         entry.insert(status);
+                        if old_status == ConsensusTxStatus::FastpathCertified {
+                            // Only path where a status can transition out of fastpath certified.
+                            inner.fastpath_certified.remove(&pos);
```

Comment:
Maybe we could add a `debug_assert` that indeed this is non empty? I don't expect surprises but it would be good to have extra validation